templates/travis-ci/.travis.yml.tmpl

```diff
@@ -27,18 +40,27 @@ jobs:
       script:
         - |
           set -e
+          {{ if .Buildevents }}
+          STEP_START=$(date +%s)
+          STEP_SPAN_ID=$(echo before_script | sum | cut -f 1 -d \ )
+          {{ end }}
           .fogg/bin/fogg apply
           if [ "$(git status --porcelain)" != "" ]; then
             echo "git tree is dirty after running fogg apply"
             echo "ensure you run fogg apply on your branch"
             git status
             exit 1
           fi
+{{ $travisCI := . }}
 {{ range $testBucket := .TestBuckets }}
     - stage: check
       script:
         - set -e
+      {{ if $travisCI.Buildevents }}
+        - STEP_START=$(date +%s)
+        - STEP_SPAN_ID=$(echo check | sum | cut -f 1 -d \ )
+      {{ end }}
       {{- range $component := $testBucket }}
-        - pushd {{ $component.Dir }} && travis_retry make {{ $component.Command }} && popd
+        - {{ if $travisCI.Buildevents }} buildevents cmd $TRAVIS_BUILD_ID $STEP_SPAN_ID {{ $component.Dir }}.{{ $component.Command }} --{{ end }} pushd {{ $component.Dir }} && travis_retry make {{ $component.Command }} && popd
```

Comment:
I wonder if we should try to capture a separate event for each try in the travis_retry's?